common/persistence/data_interfaces.go

```diff
@@ -82,6 +82,12 @@ const (
 	// UpdateWorkflowModeBypassCurrent update workflow, without current record
 	// NOTE: current record CANNOT point to the workflow to be updated
 	UpdateWorkflowModeBypassCurrent
+	// UpdateWorkflowModeSkipCurrent update workflow, without checking or update current record
+	// This mode should only be used when we don't know if the workflow being updated is the current workflow or not.
+	// For example, when updating a closed workflow, it may or may not be the current workflow.
+	// This is similar to SetWorkflowExecution, but UpdateWorkflowExecution with this mode persists the workflow as a mutation,
+	// instead of a snapshot.
+	UpdateWorkflowModeSkipCurrent
```

Comment:
This does conflict a lot with `UpdateWorkflowModeBypassCurrent`.
Can we consider something like `UpdateWorkflowModeIgnoreCurrent`? It'll be even better if we can get rid of one of these modes all together (if possible)

Reply:
yeah makes sense. `IgnoreCurrent` is a better name.

This updateMode is part of the persistence interface so not trivial to get rid of any existing ones. 
Long term if we ever get a chance to redo our persistence interface, I think the interface can explicit ask to the caller to state which columns/rows to update and to what, including changes for current rows. That will make things much more explicit. Right now the current row concept, which is an application concept is almost entirely in persistence layer...